src/device/pf_cmina.c

```diff
@@ -73,6 +78,69 @@ static void pf_cmina_send_hello(
    }
 }
 
+/**
+ * @internal
+ * Save the IP settings (and station name) to nonvolatile memory if necessary.
+ *
+ * Compares with the content of already stored settings (in order not to
+ * wear out the flash chip)
+ *
+ * @param p_ase            In:   Settings to be saved
+ */
+static void pf_cmina_save_ase_if_modified(
+   pf_cmina_dcp_ase_t     *p_ase)
+{
+   pf_cmina_dcp_ase_t      file_ase;
+   bool                    should_save = false;
+   char                    ip_string[OS_INET_ADDRSTRLEN] = { 0 };
+   char                    netmask_string[OS_INET_ADDRSTRLEN] = { 0 };
+   char                    gateway_string[OS_INET_ADDRSTRLEN] = { 0 };
+
+   pf_cmina_ip_to_string(p_ase->full_ip_suite.ip_suite.ip_addr, ip_string);
+   pf_cmina_ip_to_string(p_ase->full_ip_suite.ip_suite.ip_mask, netmask_string);
+   pf_cmina_ip_to_string(p_ase->full_ip_suite.ip_suite.ip_gateway, gateway_string);
+
+   if (os_load_blob(PNET_FILE_INDEX_IP, &file_ase, sizeof(pf_cmina_dcp_ase_t)) == 0)
+   {
+      if (memcmp(&file_ase.full_ip_suite.ip_suite, &p_ase->full_ip_suite.ip_suite, sizeof(p_ase->full_ip_suite.ip_suite)) != 0)
+      {
+         should_save = true;
+      }
+      else if (strcmp(file_ase.name_of_station, p_ase->name_of_station) != 0)
+      {
+         should_save = true;
+      }
+
+      if (should_save == true)
+      {
+         LOG_DEBUG(PF_DCP_LOG,"CMINA(%d): Updating nvm stored IP settings. "
+            "IP: %s Netmask: %s Gateway: %s Station name: %s\n", __LINE__,
+            ip_string, netmask_string, gateway_string, p_ase->name_of_station);
+      }
+   }
+   else
+   {
+      LOG_DEBUG(PF_DCP_LOG,"CMINA(%d): First nvm saving of IP settings. "
+         "IP: %s Netmask: %s Gateway: %s Station name: %s\n", __LINE__,
+         ip_string, netmask_string, gateway_string, p_ase->name_of_station);
+      should_save = true;
+   }
+
+   if (should_save == true)
+   {
+      if (os_save_blob(PNET_FILE_INDEX_IP, p_ase, sizeof(pf_cmina_dcp_ase_t)) != 0)
+      {
+         LOG_ERROR(PF_DCP_LOG,"CMINA(%d): Failed to store nvm IP settings.\n", __LINE__);
+      }
+   }
+   else
+   {
+      LOG_DEBUG(PF_DCP_LOG,"CMINA(%d): No storing of nvm IP settings (no changes). "
+         "IP: %s Netmask: %s Gateway: %s Station name: %s\n", __LINE__,
+         ip_string, netmask_string, gateway_string, p_ase->name_of_station);
+   }
+}
+
 int pf_cmina_set_default_cfg(
```

Comment:
Not so important, but maybe function name is misleading. Function is resetting config parameters according to reset mode parameter. It is not only applying a default configuration.